### ISSUE TITLE
enroll secret invalid only on Windows

### ISSUE THREAD

**Issue by danieljdaily** [kolide/fleet#1816]
### What version of `fleet` are you using (`fleet version --full`)?

fleet - version 2.0.0-rc1
  branch:       master
  revision:     a172540b7bc6a11ca261749040c9585eba370cdd
  build date:   2018-05-23T19:27:13Z
  build user:   marpaia
  go version:   go1.10.2

### What operating system are you using?

Windows 10 Enterprise for device
Red Hat 7.5 for kolide server

### What did you do?

I have installed OSquery with remote settings to connect to my on prem Kolide server.  I have tried the last two versions of osquery for windows via the MSI and chocolately, both experience the same error.  When I use the same configurations minus the file path locations for linux, the devices connect without issue.

### What did you expect to see?

I expected the devices to use my enroll secret to enroll into my Kolide instance.

### What did you see instead?

I get the following error with --tls_dump and --verbose on:

#########################
I0605 14:21:14.813107 12392 tls.cpp:241] TLS/HTTPS POST request to URI: fleet:8080/api/v1/osquery/enroll
E0605 14:21:14.838191 12392 init.cpp:574] Cannot activate tls logger plugin: No node key, TLS logging disabled.
{"enroll_secret":"hexcode","host_identifier":"device name","platform_type":"2","host_details":{"os_version":{"build":
"15063","codename":"Windows 10 Enterprise","major":"10","minor":"0","name":"Microsoft Windows 10 Enterprise","platform":"windows","platform_like":"windows","version":"10.
0.15063"},"osquery_info":{"build_distro":"windows10","build_platform":"windows","config_hash":"3a118e5b8af1688b7fb124ace8f025d3e8aa93db","config_valid":"1","extensions":"
active","instance_id":"3fdfc016-074c-4968-8247-c81907a7ca00","pid":"15592","start_time":"1528222864","uuid":"uuis","version":"3.2.6","watc
her":"11540"},"platform_info":{"revision":"1.15","vendor":"Dell Inc.","version":"1.15.4"},"system_info":{"computer_name":"device name","cpu_brand":"Intel(R) Core(TM) 
i7-6820HQ CPU @ 2.70GHz\u0000\u0000\u0000\u0000\u0000\u0000\u0000","cpu_logical_cores":"8","cpu_microcode":"194","cpu_physical_cores":"4","cpu_subtype":"-1","cpu_type":"x
86_64","hardware_model":"Precision 7710","hardware_serial":"HJQFM72","hardware_vendor":"Dell Inc.","hardware_version":"-1","hostname":"device name","local_
hostname":"device name","physical_memory":"17069719552","uuid":"uuid"}}}
{
  "error": "invalid enroll secret",
  "node_invalid": true
}
############################

I have the secret copied from my kolide server in a file (copying multiple times, even getting it from the same file on my linux server), and the config points to the file with the parameter "enroll_secret_path".  I have tried to put a bad secret in, which provides the same error but with a different Java unicode.  I have updated Kolide from 1.0.6 to 2.0.  I have verified permissions on the file containing the secret by verifying the local system user has full control and is the owner of the file.  

Here is my flag file:

--enroll_secret_path=C:\ProgramData\osquery\osquery.key
--tls_server_certs=C:\ProgramData\osquery\certs\fleet_8080.pem
--tls_hostname=fleet:8080
--host_identifier=hostname
--enroll_tls_endpoint=/api/v1/osquery/enroll
--config_plugin=tls
--config_tls_endpoint=/api/v1/osquery/config
--config_tls_refresh=10
--disable_distributed=false
--distributed_plugin=tls
--distributed_interval=3
--distributed_tls_max_attempts=3
--distributed_tls_read_endpoint=/api/v1/osquery/distributed/read
--distributed_tls_write_endpoint=/api/v1/osquery/distributed/write
--logger_plugin=tls
--logger_tls_period=10
--logger_tls_endpoint=/api/v1/osquery/log

Thanks in advance!

**Comment by groob** [comment]
I've seen users have issue with windows if the encoding/invisible chars when storing the enroll secret in a file. 
what is the contents of `C:\ProgramData\osquery\osquery.key` ? can you make sure it's plain text and that there are no newline characters?

**Comment by danieljdaily** [comment]
I have created the file with notepad and notepad ++, copying only the enroll secret directly from the Kolide fleet server via the "add new hosts" popup.  I finally decided that I would just copy the file I created for my linux servers directly to the machine, which has fixed the issue.  

I must have recreated that file a dozen times or more, but using the file from my linux machine fixed the issue.  Thank you for putting me on the path @groob !
